lib/cli/watch-output-cache.js

```diff
@@ -1,46 +1,73 @@
 /**
  * @author YJ Yang
- * @copyright 2015 YJ Yang
+ * @copyright 2015 Titus Wormer
  * @license MIT
- * @module mdast:cli:watch:output-cache
- * @fileoverview Maintain a cache of outputs to files that are being watched
- *  and were also going to be written to.
+ * @module mdast:cli:watch-output-cache
+ * @fileoverview Cache changed files which are also watched.
  */
 
 'use strict';
 
-var debug = require('debug')('mdast:cli:watch:output-cache');
+/*
+ * Dependencies.
+ */
+
+var debug = require('debug')('mdast:cli:watch-output-cache');
 var fs = require('fs');
 
 /**
- * This object keeps a mapping from a VFile's #filePath() to itself.
- * The point of using the paths as keys in this object is so that, in
- * #writeAll(), we write to each file in the cache just once.
- * @type {{}}
+ * Construct a new cache.
+ *
+ * @example
+ *   var cache = new Cache();
+ *
+ * @constructor
+ * @class {Cache}
  */
-var cache = {};
+function Cache() {
+    /**
+     * Map of file-path’s to files, containing changed files
+     * which are also watched.
+     *
+     * @member {Object}
+     */
+    this.cache = {};
+
+    /**
+     * Number of cached files.
+     *
+     * @member {number}
+     */
+    this.length = 0;
+}
 
 /**
  * Add a VFile to the cache so that its content can later be used.
  *
+ * @this {Cache}
  * @param {VFile} file - The file to add to the cache.
  */
 function add(file) {
-    var filePath = file.filePath();
+    var filePath = file.sourcePath || file.filePath();
```

Comment:
Hmm, sucks that there’s no coverage for the CLI yet.
Anyway, I’m thinking now that this isn’t necessary: `sourcePath` isn’t available because the files haven’t been added to the `fileSet` yet...